packages/flame/lib/src/components/text_component.dart

```diff
@@ -19,35 +33,34 @@ class TextComponent<T extends TextRenderer> extends PositionComponent {
   }
 
   T get textRenderer => _textRenderer;
-
+  T _textRenderer;
   set textRenderer(T textRenderer) {
     _textRenderer = textRenderer;
     updateBounds();
   }
 
-  TextComponent({
-    String? text,
-    T? textRenderer,
-    super.position,
-    super.size,
-    super.scale,
-    super.angle,
-    super.anchor,
-    super.children,
-    super.priority,
-  })  : _text = text ?? '',
-        _textRenderer = textRenderer ?? TextRenderer.createDefault<T>() {
-    updateBounds();
-  }
+  TextElement? _textElement;
 
   @internal
   void updateBounds() {
-    final expectedSize = textRenderer.measureText(_text);
-    size.setValues(expectedSize.x, expectedSize.y);
+    if (_textRenderer is FormatterTextRenderer) {
```

Comment:
Can't we put this logic inside `FormatterTextRenderer` somehow?

It doens't seems right to put logic of a specific renderer on the `TextComponent`

Reply:
We have 3 `TextRenderer` implementations right now, and I had an idea for a fourth, but all of them are `FormatterTextRenderer`s. Ideally, we would merge `FormatterTextRenderer` API into the `TextRenderer`, but that would be a breaking change for anyone who implements their own `TextRenderer`.

I would argue in favor of merging those two classes anyways, if the team agrees. But that ought to be in a separate PR, at which point we can simply remove the "else" part of this if-statement.

Reply:
I vote for merging them too, I don't think there are many (if anyone at all) that has made their own `TextRenderer` implementation.